src/libgit2/refdb_fs.c

```diff
@@ -863,6 +867,49 @@ static int iter_load_loose_paths(refdb_fs_backend *backend, refdb_fs_iter *iter)
 			error = git_vector_insert(&iter->loose, ref_dup);
 	}
 
+	if (!error && git_repository_is_worktree(backend->repo) == 1) {
+		git_iterator_free(fsit);
```

Comment:
I think from a layering perspective, the filesystem refdb is not the right place for this — that would mean that _every_ reference db would have to handle the per-corktree references themselves. I think that we should hoist the worktree logic into `refs.c` instead. In theory, the reference database should be able to operate without any knowledge of a repository (indeed, should be able to operate without a repository instance at all).

Reference databases should just handle the storage of references, and the `git_reference` family of functions should handle the business logic, including understanding whether there's a worktree or not.

Let me know if this makes sense — I can take a stab at an implementation on the references if you'd like.

Reply:
Good point.

However, here, both parts are within the filesystem.

Reply:
Not sure that I understand? Are you saying that since worktrees are in the filesystem and the default git-compatible ref database is also on the filesystem, that this is a reasonable separation of concerns?

Reply:
Er, no, you're saying that the worktree reference storage living in the commondir is actually an implementation detail of the git-compatible ref database implementation itself. 🤔 

That's a bit disappointing, but I think that you may be correct given our current database architecture.